server: remove COCKROACH_METRICS_SAMPLE_INTERVAL

tbg opened the pull request:
Hard-code it to 10s instead, the only value we actually use.

Fixes #20310.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/20810)
<!-- Reviewable:end -->

bdarnell commented:
LGTM, but we have [documented this](https://www.cockroachlabs.com/docs/stable/operational-faqs.html#why-is-disk-usage-increasing-despite-lack-of-writes), so if we remove it it should go in the release notes. We need to point to the timeseries truncation cluster setting as an alternative.

a-robinson reviewed (approved):
LGTM, but ditto on @bdarnell's comments.

tbg committed:
server: remove COCKROACH_METRICS_SAMPLE_INTERVAL

Hard-code it to 10s instead, the only value we actually use.

Fixes #20310.

Release note (general change): removed COCKROACH_METRICS_SAMPLE_INTERVAL. Users
that relied on it should reduce the value for the
`timeseries.resolution_10s.storage_duration` instead.

tschottdorf commented:
TFTR! Added the release note. Will adjust the docs, too, before merging this.

tschottdorf commented:
Docs were updated already, it turns out: https://github.com/cockroachdb/docs/blob/master/v1.2/operational-faqs.md#why-is-disk-usage-increasing-despite-lack-of-writes
